src/hotspot/share/runtime/jfieldIDWorkaround.hpp

```diff
@@ -84,12 +84,12 @@ class jfieldIDWorkaround: AllStatic {
     uintptr_t as_uint = (uintptr_t) id;
     return ((as_uint & checked_mask_in_place) != 0);
   }
-  static intptr_t raw_instance_offset(jfieldID id) {
+  static int raw_instance_offset(jfieldID id) {
```

Comment:
This doesn't look right as we've gone from returning a 64-bit value to a 32-bit value. If it is meant to be only 32-bit I expect the callers need some adjustment too.